File: chatgpt-generated/ruby/sql_injection.rb
require 'sinatra'
require 'sqlite3'

db = SQLite3::Database.new 'users.db'
db.execute 'CREATE TABLE IF NOT EXISTS users (username TEXT, password TEXT)'
db.execute 'INSERT INTO users (username, password) VALUES (?, ?)', ['admin', 'adminpass']

get '/login' do
  username = params['username']
  query = "SELECT * FROM users WHERE username = '#{username}'"
  result = db.execute(query)

  "Query: #{query}<br>" +
  result.map { |row| "User: #{row[0]}" }.join("<br>")
end

File: chatgpt-generated/php/sql_injection.php
<?php
$db = new SQLite3('users.db');
$db->exec("CREATE TABLE IF NOT EXISTS users (username TEXT, password TEXT)");
$db->exec("INSERT INTO users (username, password) VALUES ('admin', 'adminpass')");

$username = $_GET['username'] ?? '';
$query = "SELECT * FROM users WHERE username = '$username'";
$result = $db->query($query);

echo "<h2>Query:</h2> <pre>$query</pre>";
echo "<h2>Result:</h2>";
while ($row = $result->fetchArray()) {
    echo "Username: " . $row['username'] . "<br>";
}
?>
